cmd/apps/start.go

```diff
@@ -84,26 +85,48 @@ func start[B firecore.Block](cmd *cobra.Command, dataDir string, args []string,
 	}()
 	dmetering.SetDefaultEmitter(eventEmitter)
 
-	blockIDEncoding := pbfirehose.InfoResponse_BLOCK_ID_ENCODING_UNSET
-	if enc := sflags.MustGetString(cmd, "advertise-block-id-encoding"); enc != "" {
-		v, found := pbfirehose.InfoResponse_BlockIdEncoding_value[enc]
+	chainName := sflags.MustGetString(cmd, "advertise-chain-name")
+	aliases := sflags.MustGetStringSlice(cmd, "advertise-chain-aliases")
+	blockIdEncoding := sflags.MustGetString(cmd, "advertise-block-id-encoding")
+
+	blockIDEncodingEnum := pbfirehose.InfoResponse_BLOCK_ID_ENCODING_UNSET
+	firstStreamableBlock := bstream.GetProtocolFirstStreamableBlock
+
+	// If --advertise-chain-name is set, but any of the other advertise flags are not, fill them from the registry
+	if cmd.Flags().Changed("advertise-chain-name") &&
+		(!cmd.Flags().Changed("advertise-chain-aliases") ||
+			!cmd.Flags().Changed("advertise-block-id-encoding")) {
```

Comment:
@maoueh do we also want to check `advertise-block-features` here too? I don't think this is something we can extract from the registry but I might be wrong.